File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let newString = ""
  for(let i = str.length - 1; i >=0; i--){
    newString += str[i]
  }
  return newString
}
console.log("Caroline")

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("Expecting: 'eniloraC'");
  console.log("=>", reverseString("Caroline"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

//My explanation
//With buil-in methods
//Take the string and split the string into an array
//then take the contents of the array and rearrange them

//Without built-in methods
//take the last element of an array and add it to the beginning until complete (use a loop)
//take the second to last letter and place it at the end of the array successively 
//then rejoin the array back into a string 

//reversed the string with a decrementing for loop

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  split_string = str.split("")
  new_string = []
  split_string.size.times {new_string << split_string.pop}
  new_string.join 
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end
puts reverse_string('Caroline')

# Please add your pseudocode to this file
# And a written explanation of your solution

#My explanation
# Take the string and turn it into an array using the split method
#Create a new array to take in the reversed contents of the original array that was split from a string
# Iterate through each element of the array (amount of elements within the array is determined by the size and times methods)
# remove the last element of the first array (through the pop method) and shove it into the new array
# then rejoin the array into a string using the join method 